### ISSUE TITLE
ERROR: Got Http response code 401 when accessing https://api.paypal.com/v1/payments/payment. {"name":"UNAUTHORIZED_PAYMENT","message":"Unauthorized payment."

### ISSUE THREAD

**Issue by JonathanGio** [anouarabdsslm/laravel-paypalpayment#133]
**Sandbox mode,** it works perfectly, but when you switch to **LIVE mode**, it sends me this error. UNAUTHORIZED_PAYMENT , help me please!!!!

**Comment by binitghetiya** [comment]
@JonathanGio  have you changed API keys from sandbox to live ?

**Comment by JonathanGio** [comment]
@binitghetiya Yes, I already change credentials and **LIVE mode**, and even setPaymentMethod ("paypal") works correctly in LIVE MODE [](https://www.allchupe-express.com/)

**Comment by NhuLong** [comment]
I also have a problem like that TT

**Comment by ElioD** [comment]
@JonathanGio lograste resolverlo.. ?

**Comment by JonathanGio** [comment]
@ElioD  no amigo aun no... tu tienes algo al respecto!!

**Comment by NhuLong** [comment]
@JonathanGio I think paypal does not support our area .
https://developer.paypal.com/docs/classic/api/currency_codes/#creditcard

**Comment by ElioD** [comment]
@JonathanGio @NhuLong yes, logre conectarme con todas las formas de paypal, el express, el creditcard y paypal plus.

**Comment by ElioD** [comment]
Hola @JonathanGio claro, tienes algún proyecto en el cual lo quieras aplicar y ver como lo tienes para ver que esta fallando.. ?

**Comment by ElioD** [comment]
@JonathanGio deja lo checo.. haber que onda

**Comment by ElioD** [comment]
Hola @JonathanGio acabo de ver tu código las credenciales son no se declaran así, tienes que ponerlo así en config/paypal_paymant:


<?php

return [
    # Define your application mode here
    'mode' => 'sandbox',

    # Account credentials from developer portal
    'account' => [
        'client_id' => env('CLIENTE', ''),
        'client_secret' => env('SECRET', ''),
    ],

    # Connection Information
    'http' => [
        'connection_time_out' => 30,
        'retry' => 1,
    ],

    # Logging Information
    'log' => [
        'log_enabled' => true,

        # When using a relative path, the log file is created
        # relative to the .php file that is the entry point
        # for this request. You can also provide an absolute
        # path here
        'file_name' => '../PayPal.log',

        # Logging level can be one of FINE, INFO, WARN or ERROR
        # Logging is most verbose in the 'FINE' level and
        # decreases as you proceed towards ERROR
        'log_level' => 'FINE',
    ],
];

Esas variables las declaras global mente en el arvico .env de aplicación así : 

CLIENTE= ******************
SECRET= ***************
  
Tienes que checar tus credenciales en ttps://developer.paypal.com, y crear unas en My Apps & Credentials, REST API apps por que si están mal acabo de poner las mías con esos cambios y paso la transacción sin ningún problema.

Saludos.

**Comment by gutofurlan** [comment]
Resolve problem ? 
I have error code 401

**Comment by JonathanGio** [comment]
@ElioD Tendras una copia de tu repo para ver si es mi cuenta ya intente de todo...

**Comment by marteagam** [comment]
Hola amigos creo que acabo de encontrar el problema, leyendo la documentacion es necesario activar el uso para pagos con tarjeta para ello hay que ir a nuestra cuenta en www.paypal.com **-> Profile -> Profile and settings -> My selling tools -> API access**  click en **"Update"**, nos aparecera una lista en ella buscamos **"REST API integration"** y damos click en **"Manage REST API apps and credentials"**.

En la siguiente pagina nos dice claro "All REST API features are available in your sandbox account while testing. Some features will need more information when you enable them for live transactions."(Claramente no damos cuenta porque en modo sandbox todo funciona perfecto) Justo al lado hay un boton **"View permissions"** nos muestra un modal con una lista de opciones en ella veremos que "Direct cards" dice **enable** deberemos dar click y seguir los pasos siguientes para habilitar la opcion.

Les dejo el enlace directo a la ultima pagina que menciono: https://www.paypal.com/businessmanage/credentials/apiAccess/applications

**Comment by JonathanGio** [comment]
@marteagam  no amigo hasta ahora e realizado todo lo que hiciste y no amigo, tu ya pudiste solucionarlo

**Comment by ErickZH** [comment]
@JonathanGio Cambien el `mode` de la aplicación a `live` en el archivo de configuración `config/paypal_payment`. Agregue variables de entorno :nerd_face: 

```php
return [
    # Define your application mode here
    'mode' => env('PAYPAL_MODE', 'sandbox'),

    # Account credentials from developer portal
    'account' => [
        'client_id' => env('PAYPAL_CLIENT_ID', ''),
        'client_secret' => env('PAYPAL_CLIENT_SECRET', ''),
    ],

    # Connection Information
    'http' => [
        'connection_time_out' => 30,
        'retry' => 1,
    ],

    # Logging Information
    'log' => [
        'log_enabled' => env('PAYPAL_LOG', true),

        # When using a relative path, the log file is created
        # relative to the .php file that is the entry point
        # for this request. You can also provide an absolute
        # path here
        'file_name' => '../PayPal.log',

        # Logging level can be one of FINE, INFO, WARN or ERROR
        # Logging is most verbose in the 'FINE' level and
        # decreases as you proceed towards ERROR
        'log_level' => 'FINE',
    ],
];
